test/testMinsky.cc

```diff
@@ -1277,3 +1282,613 @@ TEST(TensorOps, evalOpEvaluate)
+      saveGroupAsFile(*model, groupFile);
+      
+      clearAllMaps();
+      size_t itemsBefore = model->items.size();
+      
+      insertGroupFromFile(groupFile);
+      EXPECT_GE(model->items.size(), itemsBefore);
+      
+      remove(groupFile.c_str());
+    }
+
+    // Test makeVariablesConsistent
+    TEST_F(MinskySuite, makeVariablesConsistent)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "consistVar"));
+      
+      // Just verify it doesn't crash
+      makeVariablesConsistent();
+    }
+
+    // Test garbageCollect
+    TEST_F(MinskySuite, garbageCollect)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "gcVar"));
+      
+      // Just verify it doesn't crash
+      garbageCollect();
+    }
+
+    // Test imposeDimensions
+    TEST_F(MinskySuite, imposeDimensions)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "dimVar"));
+      
+      // Just verify it doesn't crash
+      imposeDimensions();
+    }
+
+    // Test cycleCheck
+    TEST_F(MinskySuite, cycleCheck)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "cycleVar1"));
+      auto var2 = model->addItem(VariablePtr(VariableType::flow, "cycleVar2"));
+      auto op1 = model->addItem(OperationPtr(OperationType::add));
+      
+      model->addWire(var1->ports(0), op1->ports(1));
+      model->addWire(op1->ports(0), var2->ports(1));
+      
+      EXPECT_FALSE(cycleCheck());
+    }
+
+    // Test checkEquationOrder
+    TEST_F(MinskySuite, checkEquationOrder)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "orderVar"));
+      auto op1 = model->addItem(OperationPtr(OperationType::time));
+      model->addWire(op1->ports(0), var1->ports(1));
+      
+      constructEquations();
+      EXPECT_TRUE(checkEquationOrder());
+    }
+
+    // Test edited flag
+    TEST_F(MinskySuite, editedFlag)
+    {
+      flags &= ~is_edited;
+      EXPECT_FALSE(edited());
+      
+      markEdited();
+      EXPECT_TRUE(edited());
+    }
+
+    // Test reset_flag
+    TEST_F(MinskySuite, resetFlag)
+    {
+      flags |= reset_needed;
+      EXPECT_TRUE(reset_flag());
+      
+      flags &= ~reset_needed;
+      EXPECT_FALSE(reset_flag());
+    }
+
+    // Test resetIfFlagged
+    TEST_F(MinskySuite, resetIfFlagged)
+    {
+      flags |= reset_needed;
+      bool result = resetIfFlagged();
+      EXPECT_FALSE(result || reset_flag());
+    }
+
+    // Test exportSchema
+    TEST_F(MinskySuite, exportSchema)
+    {
+      string schemaFile = "/tmp/test_schema.xsd";
+      // current schema is 3, crashes on default schema 1.
+      exportSchema(schemaFile,3);
+      
+      ifstream f(schemaFile);
+      EXPECT_TRUE(f.good());
+      f.close();
+      remove(schemaFile.c_str());
+    }
+
+    // Test populateMissingDimensions
+    TEST_F(MinskySuite, populateMissingDimensions)
+    {
+      // Just verify it doesn't crash
+      populateMissingDimensions();
```

Comment:
We can do a better test here. Add some variables with some dimension information, then check afterwards that those dimensions have been added to the dimension table.